### ISSUE TITLE
Swagger properties compatibility

### ISSUE THREAD

**Issue by foxycode** [apitte/openapi#4]
I had something like this in my own implementation:

```php
<?php

namespace App\ApiV2Module\Validators;

use Sunfox\Swagger\Mapping as SWG;

/**
 * @SWG\Request(
 *     method="get",
 *     path="/campaigns",
 *     summary="Get campaign list",
 *     description="The campaigns endpoint returns list of all campaigns that could be viewed by user.",
 *     externalDocs=@SWG\ExternalDocumentation(
 *         url="http://docs.ehub.apiary.io/#reference/0/get-campaign-list/get-campaign-list"
 *     ),
 *     responses={
 *         @SWG\Response(
 *             code=200,
 *             description="Campaign list",
 *             schema=@SWG\Schema(ref="#/definitions/Campaigns"),
 *             examples=@SWG\Example(
 *                 mimeType="application/json",
 *                 file="GetCampaignList.example.json"
 *             )
 *         )
 *     }
 * )
 */
final class CampaignsDefaultValidator extends AbstractValidator
{
    /**
     * @var array
     *
     * @SWG\Parameter(
     *     name="fields",
     *     in="query",
     *     description="A list of fields to return.",
     *     required=false,
     *     type="array",
     *     items=@SWG\Items(
     *         type="string",
     *         enum={"id", "type", "status", "name", "description", "dateInserted", "cookieLifetime"}
     *     )
     * )
     */
    public $fields;

    /**
     * @var string
     *
     * @SWG\Parameter(
     *     name="type",
     *     in="query",
     *     description="Campaign type.",
     *     required=false,
     *     type="string",
     *     enum={"public", "manual", "invisible"}
     * )
     */
    public $type;

    /**
     * @var string
     *
     * @SWG\Parameter(
     *     name="status",
     *     in="query",
     *     description="Campaign status.",
     *     required=false,
     *     type="string",
     *     enum={"active", "stopped", "deleted"}
     * )
     */
    public $status;

    /**
     * @var int
     *
     * @SWG\Parameter(
     *     name="page",
     *     in="query",
     *     description="Result page number.",
     *     required=false,
     *     type="integer",
     *     default=1
     * )
     */
    public $page;

    /**
     * @var int
     *
     * @SWG\Parameter(
     *     name="perPage",
     *     in="query",
     *     description="Results per page.",
     *     required=false,
     *     type="integer",
     *     default=50,
     *     minimum=1,
     *     maximum=100
     * )
     */
    public $perPage;

    /**
     * @var string
     *
     * @SWG\Parameter(
     *     name="sort",
     *     in="query",
     *     description="Sort result by field. Prepended '-' sign means descending sort.",
     *     required=false,
     *     type="string",
     *     default="name",
     *     enum={
     *         "id", "-id", "name", "-name", "dateInserted", "-dateInserted",
     *         "cookieLifetime", "-cookieLifetime"
     *     }
     * )
     */
    public $sort;
}
```

It is possible to do something similar with this package to have proper parameter/field definition in final json schema? What is missing and needs to be implemented?

**Comment by f3l1x** [comment]
We should discuss it on NetteCamp, it's need deeper inview.

**Comment by trejjam** [comment]
With dev-master and #2, only one known thing that is missing are responses types.
Non-SWG (these from Appite) attributes have the advantage of auto-validation. If you use SWG attributes you need to specify thinks/things twice.

**Comment by foxycode** [comment]
@Trejjam I want to use Apitte annotations instead of mine, but there aren't all properties supported. For example, there is `enum` completely missing.

**Comment by trejjam** [comment]
I know, it is part of #2 

```php
final class EnumType implements ISchemaType
{
	public function createSchema(EndpointParameter $endpointParameter) : Schema
	{
		// some another types choosed by $endpointParameter

		return new Schema(
			[
				'type'  => 'string',
				'enum' => ['foo', 'bar'],
			]
		);
	}
}
```

**Comment by foxycode** [comment]
@Trejjam I can't see `EnumType` in you PR diff. Or you mean I can create my own type?

**Comment by trejjam** [comment]
Yes, the library doesn't know enum items (maybe automate it in some way in future?).  
So you need to create own `Mapper` and for openapi also own `SchemaType` (or factory of it in a similar way like Mapper).

**Comment by foxycode** [comment]
@Trejjam Will look into it. Thanks

**Comment by mabar** [comment]
Would you like something like that?
`@RequestParameter(name=foo, type=enum, allowedValues=foo,bar,baz)`

**Comment by mabar** [comment]
Current dev version don't support custom types, so let me know if you need more types (enum and list are planned) and what types is it.
Custom types could be teoretically returned. I just needed check if type is not invalid.

**Comment by foxycode** [comment]
@mabar It's not good idea to use `enum` as `type`. You need to specify type of enum values, which is `string`. Look for my example at the top.

**Comment by mabar** [comment]
I see. So `@RequestParameter(name=foo, type=string, allowedValues=foo,bar,baz)`

**Comment by foxycode** [comment]
@mabar Why `allowedValues`? Why not `enum` (which is official name in spec) as `array`? What if I need to have value containing comma?

**Comment by mabar** [comment]
Ok, enum. I didn't notice it is specified in schema.
I've never seen api which would have parameters with comma so I didn't think about it yet. Do you have idea, how to support it?

**Comment by mabar** [comment]
Problem is I would also like to support list of values. So resulting annotation would be `@RequestParameter(name=foo, type=string, list=true, enum="foo,","foo","bar","baz")`.
`example.com?foo=foo,bar,baz` is comma list separator? Or part of enum value?

**Comment by trejjam** [comment]
I think more safe way is `@RequestParameter(name=foo, type=string[], enum=["foo,","foo","bar","baz"])` and `example.com?foo[]=foo&foo[]=bar&foo[]=baz`.

**Comment by mabar** [comment]
Ok, I agree 👍

**Comment by foxycode** [comment]
I'm not sure why I used `{}` instead of `[]`. Maybe it was Doctrine Annotations limitation. Looks ok for me, so I agree too.

Look also on `items` in my example please.

**Comment by mabar** [comment]
Openapi schema is complete and enums could be defined in it. Validation on apitte side is addressed in apitte/core#49, so closing that issue
